projects/storefrontlib/src/cms-components/configurator/commons/config-attribute-types/config-attribute-numeric-input-field/config-attribute-numeric-input-field.component.ts

```diff
@@ -0,0 +1,190 @@
+import {
+  formatNumber,
+  getLocaleNumberSymbol,
+  NumberSymbol,
+} from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  OnInit,
+  Output,
+} from '@angular/core';
+import { AbstractControl, FormControl, ValidatorFn } from '@angular/forms';
+import { Configurator, LanguageService } from '@spartacus/core';
+import { ConfigFormUpdateEvent } from '../../config-form/config-form.event';
+import { ConfigUIKeyGeneratorService } from '../../service/config-ui-key-generator.service';
+
+@Component({
+  selector: 'cx-config-attribute-numeric-input-field',
+  templateUrl: './config-attribute-numeric-input-field.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ConfigAttributeNumericInputFieldComponent implements OnInit {
+  attributeInputForm: FormControl;
+  numericFormatPattern;
+  locale: string;
+
+  constructor(
+    public uiKeyGenerator: ConfigUIKeyGeneratorService,
+    public languageService: LanguageService
+  ) {}
+
+  @Input() attribute: Configurator.Attribute;
+  @Input() group: string;
+  @Input() ownerKey: string;
+
+  @Output() inputChange = new EventEmitter<ConfigFormUpdateEvent>();
+
+  mustDisplayValidationMessage(): boolean {
+    const wrongFormat: boolean =
+      (this.attributeInputForm.dirty || this.attributeInputForm.touched) &&
+      this.attributeInputForm.errors?.wrongFormat;
+
+    return wrongFormat;
+  }
+
+  determineCurrentLocale() {
+    //locales are available as languages in the commerce backend
+    this.languageService
+      .getActive()
+      .subscribe((language) => (this.locale = language))
+      .unsubscribe();
+  }
+
+  compilePatternForValidationMessage(
+    decimalPlaces: number,
+    totalLength: number
+  ) {
+    let input: string = 10 ** totalLength - 1 + '';
+    if (decimalPlaces > 0) {
+      input =
+        input.substring(0, totalLength - decimalPlaces) +
+        '.' +
+        input.substring(totalLength - decimalPlaces, totalLength);
+    }
+    const inputAsNumber: number = Number(input);
+    this.numericFormatPattern = formatNumber(
+      inputAsNumber,
+      this.locale,
+      '1.' + decimalPlaces + '-' + decimalPlaces
+    ).replace(/9/g, '#');
+  }
+
+  ngOnInit() {
+    this.determineCurrentLocale();
```

Comment:
The locale can change during the lifetime of the component (when user switchers language using the dropdown in the left top corner of the page). What do you thing about a subscription in here that updates `this.locale` whenever `langService.getActive()` emits a new value?
Note: Ideally we like to avoid subscriptions (i.e. by exposing direct observables and piping logic to them + using `async` pipe in templates). But when we really need a value sunchronously, it's justified.

Reply:
Yes, agree. Didn't see this so far as we don't have the language switcher visible when our page is displayed, but you are right we should be prepared for that.
In this case we need the subscription as the backing service would be too complex than, makes sense to have one place where the actual locale is extracted